Initial Logic commit

fedacking opened the pull request:
Added the basic logic of the game of life, with a
structure for the data, and an update function.
All of that is present in the gol.rs file, the
main file just has some basic test code.

fedacking committed:
Initial Logic commit

Added the basic logic of the game of life, with a
structure for the data, and an update function.
All of that is present in the gol.rs file, the
main file just has some basic test code.

fedacking committed:
Added test

Added multiple tests checking the basic rules of the Game of Life

cdiielsi commented on src/main.rs:
```diff
@@ -1,3 +1,22 @@
+        [D, D, D, D, D],
+    ]);
+    let _default_gol: GameOfLife<5, 5> = GameOfLife::new();
+    gol.pretty_print();
```
Maybe print a line in-between so you can easily distinguish the different boards.

fedacking replied:
Added

cdiielsi reviewed (approved):
Looks good!

DiegoCivi commented on src/gol.rs:
```diff
@@ -0,0 +1,176 @@
+            _ => CellState::Dead,
+        },
+    }
+}
```
You can also do this so you dont have anidated matches.

```rust
fn update_cell(state: CellState, neighbours: usize) -> CellState {
    match (state, neighbours) {
        (CellState::Alive, 2..3) => CellState::Alive,
        (CellState::Dead, 3) => CellState::Alive,
        _ => CellState::Dead
    }
}
```

fedacking replied:
Thanks, added the suggestion

DiegoCivi reviewed (approved):
Looks good! You could also add documentation and some comments to explain decisions.

fedacking committed:
Fixes from comments

- Added a println to divide the gol boards in the main program
- Used more comparct formatting instead of nested matches for readability.
